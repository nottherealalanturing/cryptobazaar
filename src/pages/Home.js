import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Crypto, Herocrypto } from '../components';
import { getData, selectFilteredCrypto } from '../redux/crypto/cryptoSlice';
import '../index.css';

const Home = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getData());
  }, []);
  const cryptoList = useSelector(selectFilteredCrypto);

  return (
    <div style={{ padding: '10px 10px 0 10px' }}>
      <Herocrypto
        name="Bitcoin"
        id="bitcoin"
        url="bitcoin"
        symbol="btc"
        value="20827.4673277614199303"
      />
      <span style={{ display: 'flex', color: '#dbdffd' }}>Stats by Rank</span>
      <div className="cryptoList">
        {cryptoList.slice(1).map((crypto) => (
          <Crypto
            name={crypto.name}
            id={crypto.id}
            url={crypto.id}
            symbol={crypto.symbol.toLowerCase()}
            value={crypto.priceUsd}
            key={crypto.id}
          />
        ))}
      </div>
    </div>
  );
};

export default Home;
