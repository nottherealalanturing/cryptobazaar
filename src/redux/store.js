import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cryptoReducer from './crypto/cryptoSlice';
import searchReducer from './search/searchSlice';

const reducers = {
  crypto: cryptoReducer,
  search: searchReducer,
};

const store = configureStore({
  reducer: reducers,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
