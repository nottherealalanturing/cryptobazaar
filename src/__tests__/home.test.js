import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../pages/Home';
import store from '../redux/store';

it('Home Components Renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
