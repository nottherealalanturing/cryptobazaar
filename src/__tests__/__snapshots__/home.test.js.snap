// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components Renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "10px 10px 0 10px",
    }
  }
>
  <a
    className="link"
    href="#/currency/bitcoin"
    onClick={[Function]}
  >
    <div
      className="container"
      style={
        Object {
          "backgroundImage": "url(\\"https://assets.coincap.io/assets/icons/btc@2x.png\\")",
        }
      }
    >
      <span
        className="name"
      >
        Bitcoin
      </span>
      <span
        className="value"
      >
        $20,827.47
      </span>
    </div>
  </a>
  <span
    style={
      Object {
        "color": "#dbdffd",
        "display": "flex",
      }
    }
  >
    Stats by Rank
  </span>
  <div
    className="cryptoList"
  />
</div>
`;
